examples/hello/server.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) Bull S.A.S - 2024
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program; if
+ * not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ *
+ * BXI Low Level Team
+ *
+ */
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "portals4.h"
+#include "portals4_bxiext.h"
+
+#include "hello.h"
+
+/*
+ * this file represent a server receving a message from a client
+ */
+
+int main(int argc, char **argv)
+{
+	int res;
+	int ret;
+	ptl_handle_ni_t nih;
+	ptl_pt_index_t pti;
+	ptl_handle_eq_t eqh;
+	ptl_process_t id;
+	ptl_event_t ev;
+	ptl_handle_le_t leh;
+	long num_messages;
+	char *p;
+	char msg[PTL_EV_STR_SIZE];
+	char data[50];
+	/* represent the memory used to receive the message */
+	ptl_le_t le = { .start = data,
+			.length = sizeof(data),
+			.ct_handle = PTL_CT_NONE,
+			.uid = PTL_UID_ANY,
+			.options = PTL_LE_OP_PUT };
+
+	if (argc < 2) {
+		num_messages = 0;
+		printf("Usage: client <number of possible received messages>\n default : 0\n");
+	} else {
+		num_messages = strtol(argv[1], &p, 10);
+		if (*p != '\0') {
+			fprintf(stderr,
+				"An invalid character was found before the end of the string");
+		}
+	}
+
+	printf(" \n--- server interface --- \n \n");
+
+	ret = PtlInit();
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlInit failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	}
+
+	ret = PtlNIInit(PTL_IFACE_DEFAULT, PTL_NI_NO_MATCHING | PTL_NI_PHYSICAL, SERVER_PID, NULL,
+			NULL, &nih);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlNIInit failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto fini;
+	}
+
+	ret = PtlEQAlloc(nih, 10, &eqh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto ni_fini;
+	}
+
+	ret = PtlPTAlloc(nih, 0, eqh, 0, &pti);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlPTAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto eq_free;
+	}
+
+	/* prepare an events buffer to receive a message represented by le */

```

Comment:
comment message makes no sense !
could be: "expose receive buffer to the network"
would have preferred the le structure to be filled just before the PtlLEAppend()